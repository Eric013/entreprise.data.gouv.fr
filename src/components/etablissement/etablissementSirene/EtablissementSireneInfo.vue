<template>
  <div class="company__panel panel">
    <ul class="company__info-list">
      <div class="company__item"><div class="company__item-key">SIRET</div><div class="company__item-value"> {{ result.siret }}</div></div>
      <div class="company__item"><div class="company__item-key">SIREN</div><div class="company__item-value"> {{ result.siren }}</div></div>
      <div class="company__item"><div class="company__item-key">Clef NIC</div><div class="company__item-value"> {{ result.nic }}</div></div>
      <div class="company__item"><div class="company__item-key">Activité principale Etablissement</div><div class="company__item-value"> {{ result.activite_principale }} - {{ result.libelle_activite_principale }}</div></div>
      <div class="company__item"><div class="company__item-key">Activité principale Entreprise</div><div class="company__item-value"> {{ result.activite_principale_entreprise }} - {{ result.libelle_activite_principale_entreprise }}</div></div>
      <div class="company__item"><div class="company__item-key">Nature juridique</div><div class="company__item-value"> {{ result.libelle_nature_juridique_entreprise | ifExist }}</div></div>
      <div class="company__item"><div class="company__item-key">N° TVA Intracommunautaire</div><div class="company__item-value"> {{ tvaIntracommunautaire }}</div></div>
    </ul>
  </div>
</template>

<script>
import Filters from '@/components/mixins/filters'

export default {
  name: 'EtablissementSireneInfo',
  computed: {
    result () {
      return this.$store.getters.singlePageResultEtablissement
    },
    tvaIntracommunautaire () {
      const tvaKey = (12 + 3 * (this.result.siren % 97)) % 97
      const tvaNumber = `${tvaKey}${this.result.siren}`
      return `FR${tvaNumber}`
    }
  },
  mixins: [Filters]
}
</script>

<style lang="scss" scoped>
  .panel {
    border: 2px solid $color-lighter-blue;
    // border: 2px $color-light-pink solid;
  }
</style>
