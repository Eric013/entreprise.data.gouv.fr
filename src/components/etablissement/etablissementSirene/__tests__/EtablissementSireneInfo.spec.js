import EtablissementSireneInfo from '@/components/etablissement/etablissementSirene/EtablissementSireneInfo.vue'
import { createLocalVue, shallow, mount } from '@vue/test-utils'
import { __createMocks as createStoreMocks } from '@/store/index.js'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)
jest.mock('@/store/index.js')

describe('EtablissementPanelInfo.vue', () => {
  let storeMocks
  let wrapperEPI
  let etablissementPanelInfo

  beforeEach(() => {
    storeMocks = createStoreMocks()
  
    wrapperEPI = shallow(EtablissementSireneInfo, {
      localVue,
      store: storeMocks.store
    })
    etablissementPanelInfo = wrapperEPI.vm
  })

  test('Computed value Result returns the right store getter', () => {
    expect(etablissementPanelInfo.result).toBe(storeMocks.store.getters.singlePageResultEtablissement)
  })

  test('tvaIntracommunautaire) is correctly computed', () => {
    expect(etablissementPanelInfo.tvaIntracommunautaire).toBe('FR80833057201')
  })
})

describe('EtablissementPanelInfo.vue snapshot', () => {

  const localVue = createLocalVue()
  const storeMocks = createStoreMocks()
  const wrapperEPI = mount(EtablissementSireneInfo, {
    localVue,
    store: storeMocks.store
  })

  test('It match the snapshot', () => {
    expect(wrapperEPI.vm.$el).toMatchSnapshot()
  })
})