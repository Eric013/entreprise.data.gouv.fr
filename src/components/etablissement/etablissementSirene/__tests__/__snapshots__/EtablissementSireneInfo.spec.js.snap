// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EtablissementSireneInfo.vue snapshot It match the snapshot 1`] = `
<div
  class="company__panel panel"
>
  <ul
    class="company__info-list"
  >
    <div
      class="company__item"
    >
      <div
        class="company__item-key"
      >
        SIRET
      </div>
      <div
        class="company__item-value"
      >
         
      </div>
    </div>
     
    <div
      class="company__item"
    >
      <div
        class="company__item-key"
      >
        SIREN
      </div>
      <div
        class="company__item-value"
      >
         
      </div>
    </div>
     
    <div
      class="company__item"
    >
      <div
        class="company__item-key"
      >
        Clef NIC
      </div>
      <div
        class="company__item-value"
      >
         
      </div>
    </div>
     
    <div
      class="company__item"
    >
      <div
        class="company__item-key"
      >
        Activité principale Etablissement
      </div>
      <div
        class="company__item-value"
      >
          - 
      </div>
    </div>
     
    <div
      class="company__item"
    >
      <div
        class="company__item-key"
      >
        Activité principale Entreprise
      </div>
      <div
        class="company__item-value"
      >
          - 
      </div>
    </div>
     
    <div
      class="company__item"
    >
      <div
        class="company__item-key"
      >
        Nature juridique
      </div>
      <div
        class="company__item-value"
      >
         Non renseigné
      </div>
    </div>
     
    <div
      class="company__item"
    >
      <div
        class="company__item-key"
      >
        Numéro RNA
      </div>
      <div
        class="company__item-value"
      >
         Non renseigné
      </div>
    </div>
     
    <div
      class="company__item"
    >
      <div
        class="company__item-key"
      >
        N° TVA Intracommunautaire
      </div>
      <div
        class="company__item-value"
      >
         FRNaNundefined
      </div>
    </div>
  </ul>
</div>
`;
